### PR TITLE
add ARP implementation

### LINKED ISSUE #80: ARP Protocol Support
## Reference

- [RFC 826](https://tools.ietf.org/html/rfc826)
- [Wikipedia](https://en.wikipedia.org/wiki/Address_Resolution_Protocol)
- [arp.html](https://erg.abdn.ac.uk/users/gorry/course/inet-pages/arp.html)

Screen Shot 2020-04-09 at 11 15 44 AM

### PR DESCRIPTION
**Pull request by drunkirishcoder**
## Description

ARP implementation. closes #80.

## Type of change

- [x] New protocol

## Checklist

- [x] Associated tests
- [x] Associated documentation

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
First glance is good @drunkirishcoder, but I'll run through the spec again myself and double-check on these things later today.

File: core/src/packets/arp.rs [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+///
+/// ```
+///  0                   1                   2                   3
+///  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |         Hardware Type         |         Protocol Type         |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |    H Length   |    P Length   |         Operation Code        |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |                   Source Hardware Address
```
Comment (zeeshanlakhani) [inline core/src/packets/arp.rs:40]: isn't it sender not source typically (in standard nomenclature?)?

File: core/src/packets/arp.rs [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+//! Address Resolution Protocol.
```
Comment (zeeshanlakhani) [inline core/src/packets/arp.rs:19]: lolz that i had to remember we already had the ethertype but not the protocol.

**Review by zeeshanlakhani — COMMENTED** [review]
Minor comment really, otherwise I approve. But, we should probably use sender instead of src everywhere. That seems to be overtly common for people who know arp.

File: core/src/packets/arp.rs (hunk 1 above)
Comment (drunkirishcoder) [reply core/src/packets/arp.rs]: I must be asleep. not sure where source came from. everywhere you look it's sender.

**Review by zeeshanlakhani — APPROVED** [review]

**Review by zeeshanlakhani — APPROVED** [review]

**Pull request merged**